main.js

```diff
@@ -0,0 +1,112 @@
+        console.log('問題番号は' + quizNumber);
+        let displayQuizContentHTML = ''
+        displayQuizContentHTML = `
+        <h3>[ジャンル]${fetchText.results[quizNumber].category}</h3>
+        <h3>[難易度]${fetchText.results[quizNumber].difficulty}</h3>
+        `;
+        quizContentWrapper.innerHTML = displayQuizContentHTML;
+        messageContent.textContent = fetchText.results[quizNumber].question;
+        startQuizButton.style.display = 'none';
+        const correctAnswer = fetchText.results[quizNumber].correct_answer;
+        const incorrectAnswers = fetchText.results[quizNumber].incorrect_answers;
+        const quizSelect = shuffleArray([correctAnswer, ...incorrectAnswers]);
+        console.log(quizSelect);
+        let displayButtonWrapperHTML = ''
+        displayButtonWrapperHTML = `
+        <button id="question${quizNumber}-answer-button-1" value="${quizSelect[0]}">${quizSelect[0]}</button><br>
+        <button id="question${quizNumber}-answer-button-2" value="${quizSelect[1]}">${quizSelect[1]}</button><br>
+        <button id="question${quizNumber}-answer-button-3" value="${quizSelect[2]}">${quizSelect[2]}</button><br>
+        <button id="question${quizNumber}-answer-button-4" value="${quizSelect[3]}">${quizSelect[3]}</button>
+        `;
+        buttonWrapper.innerHTML = displayButtonWrapperHTML;
+        const buttonClick = document.querySelectorAll('[id*="answer-button"]');
+        buttonClickEvent(buttonClick, correctAnswer, fetchText);
+    } else {
+        console.log('正答数は' + correctNumber);
+        displayQuizContentHTML = ''
+        displayButtonWrapperHTML = `
+        <button id="go-to-home-button" value="ホームに戻る">ホームに戻る</button>
+        `
+        quizContentWrapper.innerHTML = displayQuizContentHTML;
+        buttonWrapper.innerHTML = displayButtonWrapperHTML;
+        const goToHomeButton = document.getElementById('go-to-home-button');
+        goToHomeButton.addEventListener('click', function (event) {
+            displayQuizContentHTML = ''
+            displayButtonWrapperHTML = ''
+            quizContentWrapper.innerHTML = displayQuizContentHTML;
+            buttonWrapper.innerHTML = displayButtonWrapperHTML;
+            messageSubject.textContent = "ようこそ";
+            messageContent.textContent = "以下のボタンをクリック";
+            startQuizButton.style.display = 'inline';
+        });
+        messageSubject.textContent = "あなたの正答数は" + correctNumber + "です";
+        messageContent.textContent = "再度チャレンジしたい場合は以下をクリック";
+
+    }
+}
+function buttonClickEvent(buttonClick, correctAnswer, fetchText) {
+    quizNumber += 1;
+    messageSubject.textContent = '問題' + quizNumber;
+    buttonClick.forEach(function (button) {
+        button.addEventListener('click', function (event) {
+            const buttonClickValue = event.target.value;
+            console.log(buttonClickValue);
+            if (buttonClickValue === correctAnswer) {
+                console.log('正解です');
+                correctNumber += 1;
```

Comment:
数字のインクリメントは以下のように省略して記述できますので覚えておいてください！
```suggestion
                correctNumber++;
```